### DISCUSSION TITLE
Simultaneous Ultrasonic & Sonic TX/RX

### DISCUSSION THREAD

**Discussion by jonathanhinterberger** [ggerganov/ggwave#49] [Ideas]
### It occurs to me that the reliability of the transmission could likely be improved by utilizing more 'channels' at once, perhaps the current ultrasonic and sonic ranges. 

For instance, let's say the ECC were spread throughout the message as opposed to at the end. And we have the lower frequency channel encode the message beginning-to-end while the upper frequency channel (ultrasonic) encodes the message end-to-beginning. 

Theoretically, this means a fraction of the message might be received properly but is still able to be decoded successfully. That is, assuming I'm not missing something in my thought process. Assuming I'm correct, this also means the speed of the tx could likely be improved due to greater resilience against noise.

**Comment by ggerganov** [comment]
Hi, thanks for the interesting ideas!
This is definitely something that can be added to `ggwave` and I agree such approaches can be used to improve both the robustness and speed of the transmissions. This is along the lines of the [Direct Sequence Spread](https://github.com/ggerganov/ggwave/discussions/4#discussioncomment-851668) idea that was proposed some time ago.
I will definitely work on adding such options in future versions of the library!

If you have any specific preference for the interface for toggling such options, make sure to share them - I still haven't given it much thought so I am open to suggestions.

**Reply by jonathanhinterberger** [reply]
Awesome! There is so much potential with this library, I can't wait to see where you take it. My thought process has been consumed with ideas surrounding optimization, so I haven't really considered the interface. I will certainly share if something comes to me!
